motor-control-module/src/main/java/org/sagebionetworks/research/motor_control_module/show_step_fragment/ShowTappingCompletionStepFragment.java

```diff
@@ -0,0 +1,146 @@
+import org.sagebionetworks.research.motor_control_module.show_step_fragment.tapping.TappingButtonIdentifier;
+import org.sagebionetworks.research.motor_control_module.show_step_fragment.tapping.TappingSample;
+import org.sagebionetworks.research.motor_control_module.step.HandStepHelper;
+import org.sagebionetworks.research.motor_control_module.step.TappingStep;
+import org.sagebionetworks.research.motor_control_module.step_view.TappingCompletionStepView;
+import org.sagebionetworks.research.motor_control_module.widget.TapCountResultView;
+import org.sagebionetworks.research.presentation.model.interfaces.StepView;
+import org.sagebionetworks.research.presentation.show_step.show_step_view_models.ShowUIStepViewModel;
+
+import java.util.List;
+
+public class ShowTappingCompletionStepFragment extends
+        ShowUIStepFragmentBase<TappingCompletionStepView, ShowUIStepViewModel<TappingCompletionStepView>,
+                TappingCompletionStepViewBinding> {
+    public static ShowTappingCompletionStepFragment newInstance(@NonNull StepView stepView) {
+        ShowTappingCompletionStepFragment fragment = new ShowTappingCompletionStepFragment();
+        Bundle args = ShowStepFragmentBase.createArguments(stepView);
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    protected int getLayoutId() {
+        return R.layout.mpower2_tapping_completion_step;
+    }
+
+    @NonNull
+    @Override
+    protected TappingCompletionStepViewBinding instantiateAndBindBinding(View view) {
+        return new TappingCompletionStepViewBinding(view);
+    }
+
+    @Override
+    public void update(TappingCompletionStepView stepView) {
+        super.update(stepView);
+        TapCountResultView leftResultView = this.stepViewBinding.getLeftResult();
+        TappingResult leftResult = this.getTappingResult(HandStepHelper.Hand.LEFT);
+        this.updateTapCountResultView(leftResultView, leftResult, R.string.tapping_completion_left_description);
+        TapCountResultView rightResultView = this.stepViewBinding.getRightResult();
+        TappingResult rightResult = this.getTappingResult(HandStepHelper.Hand.RIGHT);
+        this.updateTapCountResultView(rightResultView,  rightResult, R.string.tapping_completion_right_description);
+        // Update the text view discussing the amount of time
+        TextView timeLabel = this.stepViewBinding.getTimeLabel();
+        timeLabel.setText(R.string.tapping_completion_time_label_text);
+    }
+
+    /**
+     * Updates the given result view with the information from the given TappingResult.
+     * @param view the TapCountResultView to update.
+     * @param result the TappingResult to get the data to display from.
+     * @param descriptionStringRes the String resource to display on the description label.
+     */
+    private void updateTapCountResultView(@NonNull TapCountResultView view, @Nullable TappingResult result,
```

Comment:
I think update(TappingCompletionStepView) and this method are the only ones that should go in the Fragment